orm_lib/src/TransactionImpl.cc

```diff
@@ -41,25 +41,26 @@ TransactionImpl::~TransactionImpl()
         auto loop = connectionPtr_->loop();
         loop->queueInLoop([conn = connectionPtr_,
                            ucb = std::move(usedUpCallback_),
-                           commitCb = std::move(commitCallback_)]() {
+                           commitCb = std::move(commitCallback_),
+                           autoCommit = autoCommit_]() {
             conn->setIdleCallback([ucb = std::move(ucb)]() {
                 if (ucb)
                     ucb();
             });
             conn->execSql(
-                "commit",
+                autoCommit ? "commit" : "rollback",
                 0,
                 {},
                 {},
                 {},
-                [commitCb](const Result &) {
+                [commitCb, autoCommit](const Result &) {
                     LOG_TRACE << "Transaction committed!";
-                    if (commitCb)
+                    if (autoCommit && commitCb)
                     {
                         commitCb(true);
                     }
                 },
-                [commitCb](const std::exception_ptr &ePtr) {
+                [commitCb, autoCommit](const std::exception_ptr &ePtr) {
```

Comment:
Since this is the last time we use `commitCb`. We can move it instead of making a copy.

Reply:
Are the arguments evaluated in left-to-right order? If not, we can not use move.